bpf_assets/perf_event/perf_event.c

```diff
@@ -114,44 +130,34 @@ int sched_switch(switch_args *ctx)
     u64 val = cpu_cycles.perf_read(CUR_CPU_IDENTIFIER);
     if (((s64)val > 0) || ((s64)val < -256))
     {
-        prev = prev_cpu_cycles.lookup(&cpu);
+        prev = prev_cpu_cycles.lookup(&cpu_id);
         if (prev)
         {
             cpu_cycles_delta = val - *prev;
         }
-        prev_cpu_cycles.update(&cpu, &val);
+        prev_cpu_cycles.update(&cpu_id, &val);
```

Comment:
just a thought, would cpu_cycles/delta a good proxy of cpu frequency?

Reply:
potentially, the two perf counters can also be used to estimate frequency
https://man7.org/linux/man-pages/man2/perf_event_open.2.html
```
PERF_COUNT_HW_CPU_CYCLES
                          Total cycles.  Be wary of what happens during
                          CPU frequency scaling.
PERF_COUNT_HW_REF_CPU_CYCLES (since Linux 3.3)
                          Total cycles; not affected by CPU frequency
                          scaling.
```

Reply:
Okay, in theory we can calculate the CPU frequency as the ratio of the number of cycles per second.

I will implement it and validate it with the current CPU frequency information from sysfs.